### PR TITLE
feat: add CreateMode for CTAS in WriteRel

### PR DESCRIPTION
**Pull request by scgkiran**

### CODE REVIEW

File: proto/substrait/algebra.proto
Comment (EpsilonPrime) [inline proto/substrait/algebra.proto:599]: I don't know if it applies here but Spark also has an append mode.

https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.DataFrameWriter.mode.html

File: proto/substrait/algebra.proto
Comment (EpsilonPrime) [inline proto/substrait/algebra.proto:597]: Is there a way to separate the write behavior from the error handling?  I don't have a clear implementation in mind but where each of these three settings (WriteOp, CreateMode, OutputMode) have one dimension of behavior that they deal with.

File: proto/substrait/algebra.proto
Comment (jacques-n) [reply proto/substrait/algebra.proto:599]: Yeah, I guess append makes sense (although it isn't something that commonly exists in dbs)

Comment (scgkiran) [reply proto/substrait/algebra.proto:652]: Added append.

File: proto/substrait/algebra.proto
Comment (scgkiran) [reply proto/substrait/algebra.proto:649]: I feel this is easier to follow as it is closer to the sql.

**Review by EpsilonPrime — APPROVED** [review]

**Review by EpsilonPrime — APPROVED** [review]

**Comment by jacques-n** [comment]
@anshuldata , you merged another spec change without at least two approvals. Remember, [we need at least two SMC approvals for any spec change](https://substrait.io/governance/#substrait-voting-process)

**Comment by anshuldata** [comment]
I am really sorry for this goof up again. @scgkiran write access wasn’t working he asked me to merge after approval. I forgot again that two approvals are required for this since its a feature. I will be extra extra careful going forward.

**Comment by scgkiran** [comment]
> @anshuldata , you merged another spec change without at least two approvals. Remember, [we need at least two SMC approvals for any spec change](https://substrait.io/governance/#substrait-voting-process)

Its my bad, I missed this one.

**Pull request merged**
